README.md
Eishockey Liveticker
====================

Der Eishockey Liveticker ist ein seit 2006 im Einstaz befindliches Projekt. Es wurde ursprünglich für den Verein 100% Pro Dornbirn, dem Support Verein für den EC Dornbirn, geschrieben. Ziel war es einen möglichst einfach zu administrierenden Liveticker zu produzieren, der ohne große Aufwände zu installieren, anzupassen und zu bedienen ist.

Weiters wurde sehr großer Wert auf gute Performance gelegt. Hierfür wurde ein spezieller Caching Layer gebaut, der nur dann auf die Datenbank geht, wenn tatsächlich was geändert wurde. Gehandhabt wird das über Dateien auf dem Filesystem. 

Abhängigkeiten
--------------

Der Eishockey Liveticker verwendet folgende Libraries:

 * [smarty](http://www.smarty.net/) eine templating engine, die für die Darstellung des Tickers verwendet wird.
 * [wordpress](http://www.wordpress.org) große Teile des Datenbanklayers sind von einer frühen Version von WordPress übernommen.


Fehlende Features
-----------------

Folgende Features fehlen noch im Ticker:

 * *i18n* - Aktuell gibt es den Ticker nur in Deutsch
 * Umstellung des Admin Interfaces:
   * OOP
   * Smarty Templates
   * Bessere Spieleverwaltung inkl. Löschen von Spielen
   * Automatische Erstellung einer Archivseite
   * Bessere Verwaltung von Strafen/Spielerlisten
   
Installation
------------

Bitte die [INSTALL.txt](INSTALL.txt) konsultieren. 


CHANGELOG.txt
2010/09/26 - v 1.4.2
 - fixed a small issue in mobile.css which prevented the iphone view from
   working flawlessly when the refs box was showing.
   
2010/09/21 - v 1.4.1
 - fixed a bug in admin/index.php
 - added mobile.css for handheld devices that optimizes the content for
   being looked on mobile devices.

2010/09/06 - v 1.4
 - removed PHP 4 legacy stuff (index_old.php and fragments folder)
 - changed source files to UTF-8 encoding
 - changed formatting of php source files to reflect WordPress coding standards
   (http://codex.wordpress.org/WordPress_Coding_Standards)
 - fixed various warnings of undeclared variables
 - added the constant CLOCK_RUNNING_DIRECTION (asc|desc) that sets the clock
   running direction (season 10/11 marked the change in the Austrian League)
 - added constants.php for various numeric values in the application for better
   readability
 - switched the admin panel to use smarty as well
 - admin
   - the liveticker editing panel now has radio buttons for period selection
   - the time can now be entered in ascending or descending order and as it is
     displayed on the clock.


2009/04/13 - v 1.3.2
 - fixed the create.sql script
 
2009/02/24 - v 1.3.1
 - fixed a bug when creating a new game
 - fixed an issue with 10 minutes MISC penalties
 - updated the UPDATE.txt

2009/02/18 - v 1.3
 - general
   - converted all source files to use tabs (4) instead of spaces
   - due to a bug (thanks to Franz for pointing it out) a connection to the database
     was always created even if it wasn't required. This led to problems on heavily
     frequented livetickers or tickers running on low performance servers. The 
     DBLayer class has been rewritten to initialize itself only if needed (when query()
     is called) and has a finalizer which is called in every script after the last use.
 - admin
   - prevented a goal or penalty entry from being stored when no team is selected.
   - added the following fields to the match edit screen
     - number of spectators
     - referee lineup (up to two head and two linesman)
     - length of the overtime
 - display
   - the complete output mechanism has been rewritten in an Object Oriented way. Only
     part left is the caching mechanism. All the rest has been rewritten.
   - switched the template engine to smarty (http://www.smarty.net) for far better 
     functionality in the templates
   - now knowing about the "penaltyschießen" situation thanks to a patch supplied
     by Rene Peck
   - added the Austrian Nationalliga logo in a transparent PNG format (thanks to Roland)
   - changed the default output style to match the current style of the 100 Pro Dornbirn
     Liveticker
 
2009/01/23 - v 1.2.3
 - the ticker now also knows about the time between the periods and displays 
   different text in the pauses, after the game, etc.

2008/09/26 - v 1.2.2
 - changed the behaviour of calling the index page if there are no active matches.
   In this case the newest inactive match is chosen.

2008/09/21 - v 1.2.1
 - there was still a bug with the first player of the guest team being displayed
   concatenated to the last player of the home team. Must be a difference between
   versions as I'm sure I tested it. Nevertheless it's fixed now.

2008/08/30 - v 1.2
 - calling index.php now redirects to the oldest running ticker automatically
   thus making it possible to have a always working link to the current match
   in navigations, on pages, where ever.
 - it's possible to set a match as being currently running and therefore 
   updating automatically. By default this setting is set to true.
   -> replace <meta http-equiv="refresh" content="30"> with @{meta_refresh}
 - the match list shows which games are currently running
 - the time at which an entry in the liveticker has been created is now displayed
   in the output. This timestamp is NOT modified if the entry is modified!
   -> add @{entry_time} at the desired location
 - fixed a bug where the last player of the home team was always concatenated with
   the first player of the guest team if there was no trailing newline in the 
   players list.
   
v 1.1
 - statistics
 - ability to add links which are automatically linked
 - ability to add links to the player lists

v 1.0
 - initial version

tickerdata/templates_c/%%1C^1C7^1C750DD8%%match_list.tpl.php
<?php /* Smarty version 2.6.22, created on 2010-12-29 13:50:01
         compiled from /Users/mwidmann/dropbox/projects/liveticker/templates/admin/match_list.tpl */ ?>
<?php require_once(SMARTY_CORE_DIR . 'core.load_plugins.php');
smarty_core_load_plugins(array('plugins' => array(array('modifier', 'count', '/Users/mwidmann/dropbox/projects/liveticker/templates/admin/match_list.tpl', 3, false),array('modifier', 'date_format', '/Users/mwidmann/dropbox/projects/liveticker/templates/admin/match_list.tpl', 21, false),)), $this); ?>
<?php $_smarty_tpl_vars = $this->_tpl_vars;
$this->_smarty_include(array('smarty_include_tpl_file' => "./admin/admin_header.tpl", 'smarty_include_vars' => array()));
$this->_tpl_vars = $_smarty_tpl_vars;
unset($_smarty_tpl_vars);
 ?>
		<div class="msg"><?php echo $this->_tpl_vars['msg']; ?>
</div>
		<?php if (count($this->_tpl_vars['matches']) == 0): ?>
			Keine Spiele vorhanden - 
		<?php else: ?>
			<?php echo count($this->_tpl_vars['matches']); ?>
  Matches gefunden -
		<?php endif; ?>
		<a href="match_edit.php">neues Match erstellen</a>

		<ul>
		<?php unset($this->_sections['idx']);
$this->_sections['idx']['name'] = 'idx';
$this->_sections['idx']['loop'] = is_array($_loop=$this->_tpl_vars['matches']) ? count($_loop) : max(0, (int)$_loop); unset($_loop);
$this->_sections['idx']['show'] = true;
$this->_sections['idx']['max'] = $this->_sections['idx']['loop'];
$this->_sections['idx']['step'] = 1;
$this->_sections['idx']['start'] = $this->_sections['idx']['step'] > 0 ? 0 : $this->_sections['idx']['loop']-1;
if ($this->_sections['idx']['show']) {
    $this->_sections['idx']['total'] = $this->_sections['idx']['loop'];
    if ($this->_sections['idx']['total'] == 0)
        $this->_sections['idx']['show'] = false;
} else
    $this->_sections['idx']['total'] = 0;
if ($this->_sections['idx']['show']):

            for ($this->_sections['idx']['index'] = $this->_sections['idx']['start'], $this->_sections['idx']['iteration'] = 1;
                 $this->_sections['idx']['iteration'] <= $this->_sections['idx']['total'];
                 $this->_sections['idx']['index'] += $this->_sections['idx']['step'], $this->_sections['idx']['iteration']++):
$this->_sections['idx']['rownum'] = $this->_sections['idx']['iteration'];
$this->_sections['idx']['index_prev'] = $this->_sections['idx']['index'] - $this->_sections['idx']['step'];
$this->_sections['idx']['index_next'] = $this->_sections['idx']['index'] + $this->_sections['idx']['step'];
$this->_sections['idx']['first']      = ($this->_sections['idx']['iteration'] == 1);
$this->_sections['idx']['last']       = ($this->_sections['idx']['iteration'] == $this->_sections['idx']['total']);
?>

			<?php $this->assign('entry', $this->_tpl_vars['matches'][$this->_sections['idx']['index']]); ?>
			<li>

				<?php if ($this->_tpl_vars['entry']->running): ?>
					<span style="color: #009900; font-weight: bold">AKTIV</span>
				<?php else: ?>
					<span style="color: #990000; font-weight: bold">INAKTIV</span>
				<?php endif; ?>
				<?php echo ((is_array($_tmp=$this->_tpl_vars['entry']->matchdate)) ? $this->_run_mod_handler('date_format', true, $_tmp, "%d.%m.%Y %H:%M") : smarty_modifier_date_format($_tmp, "%d.%m.%Y %H:%M")); ?>

				<b><?php echo $this->_tpl_vars['entry']->home_name; ?>
</b> gg. <b><?php echo $this->_tpl_vars['entry']->away_name; ?>
</b>
				[<a href="match_edit.php?id=<?php echo $this->_tpl_vars['entry']->match_id; ?>
">editieren</a>] -
				[<a href="liveticker.php?id=<?php echo $this->_tpl_vars['entry']->match_id; ?>
&home_name=<?php echo $this->_tpl_vars['entry']->home_name; ?>
&away_name=<?php echo $this->_tpl_vars['entry']->away_name; ?>
">liveticker</a>] -
				[<a href="../?id=<?php echo $this->_tpl_vars['entry']->match_id; ?>
" target="_new">öffentliche URL zum Liveticker</a>]

			</li>

		<?php endfor; endif; ?>
		</ul>
	</body>
</html>
